x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/80_random_rerank_retriever.yml

```diff
@@ -8,6 +8,8 @@ setup:
       indices.create:
         index: test-index
         body:
+          settings:
+            number_of_shards: 1
```

Comment:
@pmpailis just so you know, it is possible in the CCS test cases that you end up with effectively `2 shards` which has one remote and one local. 

What makes this sensitive to shard count?

Reply:
The main issue is with the idf component of the `query_string`. Looking into `explain` for the first doc we have: 
* 2 shards case: 
```
 {
                                    "value": 0.13353139,
                                    "description": "idf, computed as log(1 + (N - n + 0.5) / (n + 0.5)) from:",
                                    "details":
                                    [
                                        {
                                            "value": 3,
                                            "description": "n, number of documents containing term",
                                            "details":
                                            []
                                        },
                                        {
                                            "value": 3,
                                            "description": "N, total number of documents with field",
                                            "details":
                                            []
                                        }
                                    ]
                                },
```
1 shard case: 
```
{
                                    "value": 0.087011375,
                                    "description": "idf, computed as log(1 + (N - n + 0.5) / (n + 0.5)) from:",
                                    "details":
                                    [
                                        {
                                            "value": 5,
                                            "description": "n, number of documents containing term",
                                            "details":
                                            []
                                        },
                                        {
                                            "value": 5,
                                            "description": "N, total number of documents with field",
                                            "details":
                                            []
                                        }
                                    ]
                                },
```
Which is why the final score computed at the end is not the same.